### PR TITLE
Re-export, `TicketFuture` complete, and Documentation 

### PR DESCRIPTION
**Pull request by m-haisham**
# Description

Some things I missed in the last PR:
- Refactor and re-export so that we don't have to import `package:flip_card/flip_card_controller.dart`.
- Make sure `TickerFuture` await completion whether cancelled or not in flip and skew.
- Improvement documentation by adding doc-strings to missing classes

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Self-Code Review
- [x] Tested on example project

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Changed the example project to contain the new feature (only if applies)

### CODE REVIEW

File: lib/src/flip_card_transition.dart [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
+import 'flip_card.dart';
+
 Widget _fill(Widget child) => Positioned.fill(child: child);
 Widget _noop(Widget child) => child;
 
+/// The transition used internally by [FlipCard]
+///
+/// You obtain more control by providing your on [Animation]
+/// at the cost of built it methods like [FlipCardState.flip]
```
Comment (ciriousjoker) [inline lib/src/flip_card_transition.dart]: typo? built-in

**Review by ciriousjoker — CHANGES_REQUESTED** [review]
Minor typo maybe, aside from that it looks good. Kind of unfortunate that git doesn't pick up on the file relocation and everything shows up as a change, but what can you do.

File: lib/src/flip_card_transition.dart (hunk 1 above)
Comment (m-haisham) [reply lib/src/flip_card_transition.dart]: Ah yes, Nice catch. Thanks

**Review by ciriousjoker — APPROVED** [review]
lgtm, sorry I missed your commit

**Comment by ciriousjoker** [comment]
@BrunoJurkovic you can build a new version

**Pull request merged**